### PR TITLE
*: rewrite the logic of set.

### PR DESCRIPTION
**Pull request by hanfei1991**
I am trying to abandon the evaluate.Eval interface.
@shenli @coocood @zimulala PTAL

### CODE REVIEW

File: plan/planbuilder.go [hunk 1]
```diff
@@ -143,6 +145,36 @@ func (b *planBuilder) buildDo(v *ast.DoStmt) Plan {
 	return p
 }
 
+func (b *planBuilder) buildSet(v *ast.SetStmt) Plan {
+	p := &Set{}
+	p.tp = "Set"
```
Comment (zimulala) [inline plan/planbuilder.go]: Unify assignment format about the `tp`, using a string or a constant.

File: plan/planbuilder.go [hunk 2]
```diff
@@ -143,6 +145,36 @@ func (b *planBuilder) buildDo(v *ast.DoStmt) Plan {
 	return p
 }
 
+func (b *planBuilder) buildSet(v *ast.SetStmt) Plan {
+	p := &Set{}
```
Comment (zimulala) [inline plan/planbuilder.go]: Use `p := &Set{tp:"Set", allocator: b.allocator}` directly.

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**